lib/src/Base/Func/openturns/PiecewiseLinearEvaluation.hxx

```diff
@@ -45,11 +45,13 @@ public:
 
   /** Parameter constructor */
   PiecewiseLinearEvaluation(const Point & locations,
-                            const Point & values);
+                            const Point & values,
+                            const Bool enableExtrapolation = ResourceMap::GetAsBool("PiecewiseLinearEvaluation-DefaultEnableExtrapolation"));
 
   /** Parameter constructor */
   PiecewiseLinearEvaluation(const Point & locations,
-                            const Sample & values);
+                            const Sample & values,
+                            const Bool enableExtrapolation = ResourceMap::GetAsBool("PiecewiseLinearEvaluation-DefaultEnableExtrapolation"));
```

Comment:
After some thoughts I think its better to not modify any constructors but add a setEnableExtrapolation instead
it will keep ctors simple in case we want to add more similar options